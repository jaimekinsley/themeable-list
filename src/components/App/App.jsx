import React from 'react';
import Characters from '../characters/Characters';
import Pagination from '../pagination/Pagination';
import { usePagination } from '../../hooks/appContext';
import Header from '../header/Header';

export default function App() {
  const { page, nextPage, prevPage } = usePagination();
  return (
    <>
      <Header />
      <Pagination page={page} nextPage={nextPage} prevPage={prevPage} />
      <Characters />
    </>
  );
}
