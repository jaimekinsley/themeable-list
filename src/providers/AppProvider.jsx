import React, { useReducer, useEffect } from 'react';
import { AppContext } from '../hooks/appContext';
import reducer, { initialState } from '../reducers/appReducer';
import { fetchCharacters } from '../services/lastAirBenderAPI';

const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    fetchCharacters(state.page)
      .then(characters => dispatch({
        type: 'SET_CHARACTERS',
        payload: characters
      }));
  }, [state.page]);

  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

export default AppProvider;
