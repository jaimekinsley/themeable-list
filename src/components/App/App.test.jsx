import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import AppProvider from '../../providers/AppProvider';

describe('App component', () => {
  it.only('renders App', () => {
    const wrapper = shallow(<AppProvider><App /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
