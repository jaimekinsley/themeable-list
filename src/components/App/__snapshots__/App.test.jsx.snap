// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "characters": null,
        "page": 1,
        "theme": "light",
      },
    }
  }
>
  <App />
</ContextProvider>
`;
